### PR TITLE
django environ initiated

### PR DESCRIPTION
**Pull request by JDRanpariya**
I have changed settings in order to use django environ and make it easy but its still having lot of issues I hope will solve them some of the problems are:

1. it's not getting values form .env file
2. I'm a bit confused about if conditions are properly written with environ
3. I did't get the full or very comprehensive tutorial on using django environ
but some of them are really help and are as follows...

1. (https://djangostars.com/blog/configuring-django-settings-best-practices/)
2. (https://django-environ.readthedocs.io/en/latest/#)

One most prominent error i am getting is   Set the SECRET_KEY environment variable.

### CODE REVIEW

**Comment by JDRanpariya** [comment]
I have divided setting.py into 3 parts django, project settings and third party settings as in 

> Two Scoops of Django: Best Practices

File: .env_sample [hunk 1]
```diff
@@ -1,19 +1,64 @@
...
-GOOGLE_MAP_API_KEY=<google_map_key>
-DJANGO_SECRET_KEY=<django_secret_key>
-PROPUBLICA_API_KEY=<pro_publica_api_key>
 
+DEBUG=on
+
+
+# database variables
+
+DATABASE_URL=postgresql://urser:un-githubbedpassword@127.0.0.1:8458/database
```
Comment (brylie) [inline .env_sample:7]: We should use environment variables for the database configuration. Use f-strings to combine the `POSTGRES_HOST`, `POSTGRES_PORT`, `POSTGRES_DB`, `POSTGRES_USER`, and `POSTGRES_PASSWORD` into a single string.

File: .env_sample [hunk 2]
```diff
@@ -1,19 +1,64 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (brylie) [inline .env_sample:64]: Only one blank line at the end of the file.

File: project/civiwiki/settings/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from .django import *
+from .project import *
+from .third_party import *
```
Comment (brylie) [inline project/civiwiki/settings/__init__.py:3]: Should be one newline at the end of the file.

File: project/civiwiki/settings/django.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            'NAME': env.str("CIVIWIKI_LOCAL_NAME"),
+            'ENGINE': 'django.db.backends.postgresql_psycopg2',
+            'USER': env.str("CIVIWIKI_LOCAL_USERNAME"),
+            'PASSWORD': env.str("CIVIWIKI_LOCAL_PASSWORD"),
+        },
+    }
+
+
+# Email Backend Setup
+if env('EMAIL_HOST'):
```
Comment (brylie) [inline project/civiwiki/settings/django.py:118]: Perhaps make this more explicit by saying `if env("EMAIL_HOST") == None:`.

File: project/civiwiki/settings/project.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# Apex Contact for Production Errors
+ADMINS = [('Development Team', 'dev@civiwiki.org')]
+
+
+# AWS S3 Setup
+
+public_root = BASE_DIR.path('public/')
+
+if env('AWS_STORAGE_BUCKET_NAME'):
```
Comment (brylie) [inline project/civiwiki/settings/project.py:50]: These falsy checks are somewhat confusing, as they rely on `env()` returning `None` for missing values.  Try to make it more clear what this is checking for.

**Comment by brylie** [comment]
@pep8speaks pep8ify

File: .env_sample (hunk 1 above)
Comment (JDRanpariya) [reply .env_sample:7]: I didn't get this are you talking about settings file or env file?

Comment (JDRanpariya) [reply .env_sample:7]: And I don't get this pep 8 thing what's that?

**Comment by brylie** [comment]
Thanks for all of your work so far @JDRanpariya :-)

**Comment by JDRanpariya** [comment]
> Thanks for all of your work so far @JDRanpariya :-)

No problem at all. Nice to meet you 
by the way still I haven't seen the website interface still struggling to run website on local server

**Comment by brylie** [comment]
Ok, I hope you are able to get things running soon.

File: .env_sample (hunk 1 above)
Comment (brylie) [reply .env_sample:7]: Sorry for the delay. Pep8 is a code linter that makes sure we follow basic code conventions.

**Comment by brylie** [comment]
Closing this issue since the code may have changed significantly lately

@all-contributors please add @JDRanpariya for code

**Pull request closed without merging**